### PR TITLE
Adds GH workflow to build docker images on version tags

### PR DESCRIPTION
**Pull request by vpereira01**
Adds a GitHub workflow, and custom docker file, using docker buildx which builds docker container images for different CPU architectures and publishes these images to GitHub packages. 

Workflow will automatically run when a version tag is created, pattern `v*`, example `v0.23.0-beta`. Also supports manual workflow trigger to allow the flexibility to build images without needing to create a new version tag.

Fixes: #1623 

Example of how packages are listed from my fork https://github.com/vpereira01/btcd/pkgs/container/btcd

Few notes:
* The step, `Extract metadata (tags, labels) for Docker`, will generate the tag **latest** when creating a tag `v0.99` on master. Example from my fork: https://github.com/vpereira01/btcd/runs/5561633723?check_suite_focus=true . Not sure if this should be kept given the comments on #1632  
* I think the packages must be made public. GitHub seems to default packages to be private so anonymous docker pulls will not be allowed.
* I have not used image digests on Dockerfile given the digest varies according to the TARGETARCH.
* The platforms `linux/amd64,linux/arm64,linux/arm,linux/386` seems good enough to cover must of uses cases. To support specific GOARM a lot rework is required.

References:
* https://www.docker.com/blog/docker-v2-github-action-is-now-ga/
* https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages
* Possible optimizations https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/

### CODE REVIEW

File: .github/workflows/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# GitHub action dockerfile
+# Requires docker experimental features as buildx and BuildKit so not suitable for developers regular use.
+# https://docs.docker.com/develop/develop-images/build_enhancements/#to-enable-buildkit-builds
+
+###########################
+# Build binaries stage
+###########################
+FROM --platform=$BUILDPLATFORM golang:1.16.15-alpine3.15 AS build
```
Comment (jcvernaleo) [inline .github/workflows/Dockerfile:8]: I think go 1.16 is no longer supported.  Really should use 17 (or 18).

File: .github/workflows/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+###########################
+# Build docker image stage
+###########################
+FROM alpine:3.15
+COPY --from=build /go/bin /bin
+# 8333  Mainnet Bitcoin peer-to-peer port
+# 8334  Mainet RPC port
+EXPOSE 8333 8334
+ENTRYPOINT ["btcd"]
```
Comment (jcvernaleo) [inline .github/workflows/Dockerfile:24]: Files really should end in a newline.

File: .github/workflows/dimagespub.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+       
+      - name: Build and push Docker images
+        uses: docker/build-push-action@ac9327eae2b366085ac7f6a2d02df8aa8ead720a
+        with:
+          file: .github/workflows/Dockerfile
+          labels: ${{ steps.meta.outputs.labels }}
+          platforms: ${{ env.TPLATFORMS }}
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
```
Comment (jcvernaleo) [inline .github/workflows/dimagespub.yml:52]: Really should end in a newline.

**Review by jcvernaleo — CHANGES_REQUESTED** [review]
Three super minor comments.

File: .github/workflows/Dockerfile (hunk 1 above)
Comment (vpereira01) [reply .github/workflows/Dockerfile]: Done.

**Review by jcvernaleo — APPROVED** [review]
Looks good now.

OK

**Pull request merged**